src/Core/Waiter.php

```diff
@@ -114,14 +114,15 @@ public function __destruct()
      * Make sure the actual request is executed.
      *
      * @param float|null $timeout Duration in seconds before aborting. When null wait until the end of execution.
+     * > When null wait until the end of execution => not what the code does, isn't it? See proposal in Result.php
      *
      * @return bool whether the request is executed or not
      *
      * @throws NetworkException
      */
     final public function resolve(?float $timeout = null): bool
     {
-        if (!isset($this->response)) {
+        if (null === $this->response) {// calling twice resolve() in a row will yield strange results I think
```

Comment:
Hm.. We currently allow that. It will throw exception twice.

Reply:
but what if there is no exception?